cli/src/api/pcluster_api.py

```diff
@@ -65,16 +65,38 @@ def __repr__(self):
         return json.dumps(self.__dict__)
 
 
-class ImageInfo:
+class ImageBuilderInfo:
     """Minimal representation of a building image."""
 
     def __init__(self, imagebuilder: ImageBuilder):
-        self.image_name = imagebuilder.stack.name
-        self.imagebuild_status = imagebuilder.imagebuild_status
-        self.stack_status = imagebuilder.stack.status
-        self.stack_arn = imagebuilder.stack.id
+        self.stack_exist = False
+        self.image_exist = False
         self.region = get_region()
-        self.version = imagebuilder.stack.version
+        # image config file url
+        self.image_configurtion = imagebuilder.config_url
+
+        if AWSApi.instance().cfn.stack_exists(imagebuilder.image_name):
```

Comment:
I'm not sure about this call in the ImageBuilderInfo, it seems to me that it's up of the caller, using ImageBuilderInfo to pass the known information.

You can do it by passing the stack to the ImageBuilder constructor:
```python
def __init__(self, imagebuilder: ImageBuilder, stack: StackInfo = None):
  ...
if stack:
    # imagebuilder stack info
    self.stack_exist = True
    ...

```

Reply:
I don't fully understand this comment, you mean in API methods to pass imagebuilder and stack to ImageBuilderInfo constructor? like
```
    @staticmethod
    def delete_image(image_name: str, region: str, force: bool = False):
            ...
            imagebuilder = ImageBuilder(image_name)
            try:
               stack = imagebuilder.stack
            except StackError:
               stack = None
            imagebuilder.delete(force=force)
            return ImageBuilderInfo(imagebuilder=imagebuilder, stack=stack)
```
?

Reply:
Yes when the api method has the stack information is up to him to pass these information to ImageBuilderInfo, we don't need to do another call into the ImageBuilderInfo to retrieve stack information.

In this way the ImageBuilderInfo is just a collector of "static" information, we can consider it as an object representing the Json dictionary of the Api response, it shouldn't have methods or do api calls.